Straw-man for our workshop (#1)
observability-lab-cse/observability-lab-workshop: 02-deploy-application/README.md

# Deploying the Application into AKS

> Goal:
>
> 1. Having application run full E2E and understanding the pieces.
>

> > 2. Observe the application with basic "instrumentation" including emitted log lines and possibly some metrics using `kubectl`. Get familiar with out-of-the-box AKS cluster observability (eg. workloads, pods etc.) Mentioning of ContainerInsights and its capabilities.

Reviewer: ```suggestion
> 2. Getting familiar with out-of-the-box AKS cluster observability (e.g. workloads, pods etc.) Mentioning of Container Insights and its capabilities.
```

Reviewer: We could still show the logs through kubectl, though I would skip the first sentence and just mention out-of-the box AKS observability. Later we can review and decide what exactly we will show.

> 1. Deploy Services to AKS

Reviewer: ```suggestion
1. Deploy the application to AKS
```

> 2. Create Devices and see temperature and state being updated

Reviewer: ```suggestion
2. Create devices and see temperature and state being updated
```

# Deploying the Application

> > Goal: Observe the application with basic "instrumentation," including emitted log lines and possibly some metrics. We aim to obtain an overview of the current state of the application, either based on the customer's existing setup or by setting up a simple monitoring system with them.

Reviewer 2: ```suggestion
> Goal: Observe the application with basic "instrumentation", including emitted log lines and possibly some metrics. We aim to obtain an overview of the current state of the application, either based on the customer's existing setup or by setting up a simple monitoring system with them.
```

> > Goal: Observe the application with basic "instrumentation," including emitted log lines and possibly some metrics. We aim to obtain an overview of the current state of the application, either based on the customer's existing setup or by setting up a simple monitoring system with them.

Reviewer 3: My understanding is that its any customer we have in the engagements. In the meaning that this section of the workshop allows the participants to learn how to obtain an overview of the current state of the application, either based on the customer's existing setup or by setting up a simple monitoring system with them.

> > Goal: Observe the application with basic "instrumentation," including emitted log lines and possibly some metrics. We aim to obtain an overview of the current state of the application, either based on the customer's existing setup or by setting up a simple monitoring system with them.

Reviewer 3: We could have this section as the first step for the instrumentation, meaning "deploy and see what is available out of the box in Azure", so in the first section (01) we deploy the infra: AKS, cosmosdb etc, in the second we deploy the applications into AKS and see how the system behaves and what we can see in Azure. Maybe spot what's missing which would be an introduction tot the next section that goes through the automatic instrumentations for the modules

> > Goal: Observe the application with basic "instrumentation," including emitted log lines and possibly some metrics. We aim to obtain an overview of the current state of the application, either based on the customer's existing setup or by setting up a simple monitoring system with them.

Author: Hope the updated version does clarify things.  If not let me know :)
